rpc/src/module/ipc.rs

```diff
@@ -0,0 +1,784 @@
+        }
+        return Err(RPCError::custom(
+            RPCError::IPC,
+            format!("Get out point failed: {:?}", script_locator),
+        ));
+    }
+}
+
+#[async_trait]
+impl IpcRpc for IpcRpcImpl {
+    fn ipc_call(
+        &self,
+        script_locator: IpcScriptLocator,
+        req: IpcRequest,
+        env: Option<IpcEnv>,
+    ) -> Result<IpcResponse> {
+        let mut dl = Resource::new(self.shared.clone());
+        let fmt = req.payload_format;
+        let env = self.ipc_env(&mut dl, self.ipc_script_data(script_locator)?, env)?;
+        let tx = packed::Transaction::from(env.tx).into_view();
+        let mut tx = resolve_transaction(tx, &mut HashSet::new(), &dl, &dl)
+            .map_err(|e| RPCError::custom_with_error(RPCError::TransactionFailedToResolve, e))?;
+        // Due to the existence of system cell cache, some system scripts will not be obtained from Resource. There is
+        // currently no way to disable the system cell cache, so manually override them here.
+        for e in tx.resolved_cell_deps.iter_mut() {
+            if let Some(CellStatus::Live(meta)) = dl.cell.get(&e.out_point) {
+                *e = meta.clone()
+            }
+        }
+        let snapshot = self.shared.cloned_snapshot();
+        let pipesctx = PipesCtx::default();
+        let script_verifier = TransactionScriptsVerifier::<_, _, Machine>::new_with_generator(
+            Arc::new(tx),
+            dl,
+            snapshot.cloned_consensus(),
+            Arc::new(TxVerifyEnv::new_submit(snapshot.tip_header())),
+            overload_ckb_syscalls,
+            pipesctx.clone(),
+        );
+        std::thread::spawn(move || {
+            // Ignore its return value as it is unimportant.
+            script_verifier.verify_single(
+                match env.script_group_type {
+                    ScriptGroupType::Lock => ckb_script::ScriptGroupType::Lock,
+                    ScriptGroupType::Type => ckb_script::ScriptGroupType::Type,
+                },
+                &env.script_hash.0.pack(),
+                // I need a maximum number of cycles so that the vm can always be stopped.
+                // Not sure if this is a sensible value.
+                snapshot.cloned_consensus().max_block_cycles,
+            )
+        });
+        // In any case, we will close all pipes after a certain period of time.
+        let pipesfin = pipesctx.clone();
+        std::thread::spawn(move || {
+            std::thread::sleep(std::time::Duration::from_secs(8));
+            pipesfin.vw.lock().unwrap().close();
+            pipesfin.vr.lock().unwrap().close();
+            pipesfin.nw.lock().unwrap().close();
+            pipesfin.nr.lock().unwrap().close();
+        });
+        let req = RequestPacket::new(
+            req.version.value() as u8,
+            req.method_id.value(),
+            match fmt {
+                IpcPayloadFormat::Hex => {
+                    let str = req
+                        .payload
```

Comment:
Do we need to check the size of `req.payload`?

Reply:
CKB's jsonrpc has set `max_request_body_size = 10M`, I don't have much idea whether we need to set the size limit additionally.

Reply:
I added defensive code to avoid the payload being too large.